data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeParser.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.IntStream;
+import org.antlr.v4.runtime.Lexer;
+import org.antlr.v4.runtime.TokenSource;
+import org.antlr.v4.runtime.TokenStream;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Handles interaction with ANTLR generated parser and lexer classes and caches results.
+ */
+@Named
+class ParseTreeParser implements Parser<ParseTree> {
+    private final Map<String, ParseTree> cache = new HashMap<>();
+    private final ParserErrorListener errorListener;
+    private final Lexer lexer;
+    private final DataPrepperExpressionParser parser;
```

Comment:
If there is nothing specifically tied to DataPrepperExpressionParser, shall we keep use generic antlr Parser here to make it consistent with Lexer?

Reply:
DataPrepperExpressionParser is needed to access `DataPrepperExpressionParser.expression()`